### PR TITLE
fix(provisioning): Option to get user without creating

### PR DESCRIPTION
**Pull request by sentaur-athena**
We have a partnership case where partner doesn't want us to create the user. In this case we just create an org member for that email and when email gets the request and approve it through the link user gets created.

### CODE REVIEW

File: src/sentry/services/hybrid_cloud/user/impl.py [hunk 1]
```diff
@@ -174,22 +174,17 @@ def get_first_superuser(self) -> Optional[RpcUser]:
...
     def get_or_create_user_by_email(
-        self, *, email: str, ident: Optional[str] = None, referrer: Optional[str] = None
-    ) -> RpcUser:
+        self,
+        *,
+        email: str,
+        ident: Optional[str] = None,
+        referrer: Optional[str] = None,
+        skip_creation: Optional[bool] = False,
+    ) -> Optional[RpcUser]:
```
Comment (markstory) [inline src/sentry/services/hybrid_cloud/user/impl.py:183]: Isn't it odd that `get_or_create` has a parameter that makes it do something totally different? Should we have a `get_user_or_none` method instead?

Comment (sentaur-athena) [reply src/sentry/services/hybrid_cloud/user/impl.py:183]: It is odd. If I get_or_none, then I have to create in a different rpc. I was trying to minimize number of rpc calls

Comment (sentaur-athena) [reply src/sentry/services/hybrid_cloud/user/impl.py:183]: I can call it get_user_by_email and add a flag that says `should_create`. That sounds better.

Comment (sentaur-athena) [reply src/sentry/services/hybrid_cloud/user/impl.py:183]: Oh got it , you're suggesting to have both.

**Review by markstory — APPROVED** [review]

**Review by dashed — APPROVED** [review]

**Pull request merged**